### DISCUSSION TITLE
如何从代码文件中启动mitmweb

### DISCUSSION THREAD

**Discussion by wobuyaotutou** [mitmproxy/mitmproxy#6877] [Q&A]
我想要学习一下代码的底层逻辑，找到了mitmweb的入口位置，但是不会从代码文件角度启动mitmweb，我想通过python启动mitmweb，感谢大佬的协助

**Comment by mhils** [comment]
We don't have anyone on the team who can moderate Chinese content, so please post in English.

**Comment by moon2l** [comment]
Try like this: PYTHONPATH=<MITMPROXY_REPO_ROOT_PATH>  python release/specs/mitmweb

Refer to the following test code:
`test_mitmweb` function in files:
<MITMPROXY_REPO_ROOT_PATH>/test/mitmproxy/tools/test_cmdline.py
<MITMPROXY_REPO_ROOT_PATH>/test/mitmproxy/tools/test_main.py
